src/Core/Settings.h

```diff
@@ -141,6 +141,7 @@ class IColumn;
     \
     M(Bool, force_index_by_date, 0, "Throw an exception if there is a partition key in a table, and it is not used.", 0) \
     M(Bool, force_primary_key, 0, "Throw an exception if there is primary key in a table, and it is not used.", 0) \
+    M(Bool, force_max_partition_limit, 0, "Throw an exception if max_partitions_to_read is violated.", 0) \
```

Comment:
I feel trobled with the fact that two settings must be set simultaneously for this to take effect.

Reply:
Ok.. Perhaps we can set `force_max_partition_limit` default to 1.

Reply:
Let's name it `max_partitions_to_read` - the same as per-table setting.
The logic will be the following:

- if query level is not `changed`, table level setting will be used;
- query level setting can override table level setting;
- even if it is explicitly set to zero, it will override table level setting as it will be `changed`.

Reply:
Done